Handle new Skyline sequentially hashed audit logs

labkey-jeckels opened the pull request:
#### Rationale
Skyline Daily updated its audit log file hashing approach, so that each entry's hash includes the hash of the previous entry

#### Changes
* Handle sequential hashed, based on file version
* Tests to validate clean and hacked audit logs
* Misc code cleanup based on Java and LabKey conventions, and IntelliJ auto-refactors

Commit:
Support sequential hashing for Skyline audit logs, as of document version 20.21

Improve consistency with Java/LabKey coding conventions

IntelliJ refactors and misc code cleanup

labkey-jeckels committed:
Add variants of audit files

Commit:
Validate acceptable and hacked sequential and non-sequential hashes

Ensure that InputStreams are always closed

More misc cleanup

labkey-ankurj commented on src/org/labkey/targetedms/SkylineAuditLogManager.java:
```diff
@@ -187,116 +184,119 @@ private boolean verifyPreRequisites(AuditLogImportContext pContext) throws Audit
+            return 0;
+        }
+
+        if(pContext._runId != null)       //set the document version id on the chronologically last log entry.
```
```suggestion
        if (pContext._runId != null)       //set the document version id on the chronologically last log entry.
```

labkey-ankurj commented on src/org/labkey/targetedms/SkylineAuditLogManager.java:
```diff
@@ -187,116 +184,119 @@ private boolean verifyPreRequisites(AuditLogImportContext pContext) throws Audit
+
+        AuditLogTree treePointer = pContext._logTree;
+        int persistedEntriesCount = 0;
+        for(AuditLogEntry ent : entries)
```
```suggestion
        for (AuditLogEntry ent : entries)
```

labkey-ankurj commented on src/org/labkey/targetedms/SkylineAuditLogManager.java:
```diff
@@ -187,116 +184,119 @@ private boolean verifyPreRequisites(AuditLogImportContext pContext) throws Audit
+                treePointer = newTreeEntry;
+            }
         }
+        if(persistedEntriesCount == 0)      //if no entries were actually saved into the database we are uploading an earlier document version
```
```suggestion
        if (persistedEntriesCount == 0)      //if no entries were actually saved into the database we are uploading an earlier document version
```

labkey-ankurj commented on src/org/labkey/targetedms/SkylineAuditLogManager.java:
```diff
@@ -442,29 +442,30 @@ public void deleteDocumentLog(@NotNull GUID pDocumentGUID)
         }
 
-        private AuditLogTree persistALogFile(String filePath, Integer runId) throws IOException, AuditLogException{
+        private AuditLogTree persistALogFile(String filePath, Integer runId) throws IOException, AuditLogException
```
should be Long runId to avoid possible exceptions

labkey-ankurj commented on src/org/labkey/targetedms/SkylineAuditLogManager.java:
```diff
@@ -442,29 +442,30 @@ public void deleteDocumentLog(@NotNull GUID pDocumentGUID)
         }
 
-        private int getNewRunId(GUID pDocumentGUID){
+        private int getNewRunId(GUID pDocumentGUID)
```
Long runId

labkey-jeckels committed:
Update src/org/labkey/targetedms/SkylineAuditLogManager.java

Co-authored-by: Ankur Juneja <ankurj@labkey.com>

labkey-jeckels committed:
Update src/org/labkey/targetedms/SkylineAuditLogManager.java

Co-authored-by: Ankur Juneja <ankurj@labkey.com>

labkey-jeckels committed:
Update src/org/labkey/targetedms/SkylineAuditLogManager.java

Co-authored-by: Ankur Juneja <ankurj@labkey.com>

Commit:
Code review feedback:

Shift from TestRun to TargetedMSRun to ensure correct ID field data type

Prefer run objects to raw long values

labkey-jeckels commented:
@labkey-ankurj There are plenty of other lines that don't use our typical code formatting conventions. We could do a separate reformatting PR at some point, but I've accepted the specific edits you suggested.

I also cleaned up some of the run ID handling.

labkey-ankurj commented:
sounds good. Thanks

labkey-jeckels committed:
Update to match new API and avoid double-encoding URL
